front/src/modules/rollingStock/hooks/useFilterRollingStock.tsx

```diff
@@ -143,10 +144,19 @@ export default function useFilterRollingStock() {
     pageSize: 1000,
   });
 
+  const usefulRollingStocks = useMemo(
+    () =>
+      isStdcm
+        ? allRollingStocks.filter((rs) => FREIGHT_ROLLING_STOCKS.includes(rs.name))
+        : allRollingStocks,
+    [allRollingStocks]
+  );
```

Comment:
will this be really triggered enough times to justify this `useMemo()`?

Reply:
Yes, otherwise it will be triggered at each re-render, so each time a filter change, the input change etc...
It should only be triggered at the first render

Reply:
just tried without it, in fact we have so many rerenders (at first 3 minimum), this part of the code seems very energy intensive

Reply:
Done :white_check_mark: